core/src/main/java/io/questdb/griffin/SqlCompilerImpl.java

```diff
@@ -1431,11 +1431,11 @@ private void compileUsingModel(SqlExecutionContext executionContext) throws SqlE
         lexer.unparseLast();
         codeGenerator.clear();
 
-        final ExecutionModel executionModel = compileExecutionModel(executionContext);
+        ExecutionModel executionModel = compileExecutionModel(executionContext);
         switch (executionModel.getModelType()) {
             case ExecutionModel.QUERY:
                 LOG.info().$("plan [q=`").$((QueryModel) executionModel).$("`, fd=").$(executionContext.getRequestFd()).$(']').$();
-                compiledQuery.of(generate((QueryModel) executionModel, executionContext));
+                compiledQuery.of(generateWithRetries((QueryModel) executionModel, executionContext));
```

Comment:
That's a good one! I thought it was dealt with only when a cached `RecordCursorFactory` was used. 
but apparently, in some cases, it's handled even when thrown while compiling. 
but it looks very inconsistent: 
1. `PGConnectionContext`: It handles when `TableReferenceOutOfDateException` is thrown while using pre-compiled factories. but it does not expect `compile()` to throw `TableReferenceOutOfDateException`. This will likely lead to an internal error. ❗ 
2. `JsonQueryProcessor`: It handles the case where `compile()` throws `TableReferenceOutOfDateException` ✅ 
3. `TextQueryProcessor` - same as `PGConnectionContext` - it handles only cached queries, but not `compile()`❗

So, I see 2 options:
1. Make `TextQueryProcessor` and `PGConnectionContext` do the same thing `JsonQueryProcessor` does: They should expect `compile()` to throw `TableReferenceOutOfDateException`. Aka external retrying. 
2. Make sure `compile()` does to throw it. Why it should? That's what this PR is trying to do. Let's call it internal retrying. 

no. 2 looks more correct to me. but it's also more risky I guess. If we go this way then we should remove the retrying from `JsonQueryProcessor` and probably also from `compileBatch()` (both well spotted, thanks!)

So I have 2 questions for you:
1. Does ☝️ make sense?
2. What path do you consider to be better?

Reply:
I'd also go with the second option.

Reply:
alright, let me have a look. and thank you for thinking about this, input like this is exactly what I was looking for!

Reply:
@puzpuzpuz it's ready for another look. lmk what you think. thanks!